docs/source/basics/multimodal_datasets.rst

```diff
@@ -0,0 +1,285 @@
+
+.. code-block:: python
+
+    # data/my_data.json
+    [
+        {
+            "dialogue": [
+                {
+                    "from": "human",
+                    "value": "<image>What time is it on the clock?",
+                },
+                {
+                    "from": "gpt",
+                    "value": "It is 10:00 AM.",
+                },
+            ],
+            "image_path": "images/clock.jpg",
+        },
+        ...,
+    ]
+
+.. code-block:: python
+
+    from torchtune.models.flamingo import FlamingoTransform
+    from torchtune.datasets.multimodal import multimodal_chat_dataset
+
+    transform = FlamingoTransform(
+        path="/tmp/Meta-Llama-3-8B-Instruct/original/tokenizer.model",
+        prompt_template="torchtune.data.QuestionAnswerTemplate",
+        max_seq_len=8192,
+        tile_size=224,
+        patch_size=14,
+    )
+    ds = multimodal_chat_dataset(
+        model_transform=model_transform,
+        source="json",
+        data_files="data/my_data.json",
+        column_map={
+            "dialogue": "conversations",
+            "image_path": "image",
+        },
+        image_dir="/home/user/dataset/",  # /home/user/dataset/images/clock.jpg
+        image_tag="<image>",
+        split="train",
+    )
+    tokenized_dict = ds[0]
+    print(transform.decode(tokenized_dict["tokens"], skip_special_tokens=False))
+    # '<|begin_of_text|><|start_header_id|>user<|end_header_id|>\n\nQuestion:<|image|>What time is it on the clock?Answer:<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\nIt is 10:00AM.<|eot_id|>'
+    print(tokenized_dict["encoder_input"]["images"][0].shape)  # (num_tiles, num_channels, tile_height, tile_width)
+    # torch.Size([4, 3, 224, 224])
+
+.. code-block:: yaml
+
+    # In config
+    model_transform:
+      _component_: torchtune.models.flamingo
+      path: /tmp/Meta-Llama-3-8B-Instruct/original/tokenizer.model
+      prompt_template: torchtune.data.QuestionAnswerTemplate
+      max_seq_len: 8192
+
+    dataset:
+      _component_: torchtune.datasets.multimodal.multimodal_chat_dataset
+      source: json
+      data_files: data/my_data.json
+      split: train
+      column_map:
+        dialogue: conversations
+        image_path: image
+      image_dir: /home/user/dataset/
+      image_tag: "<image>"
+      split: train
+
+Multimodal dataset format
+-------------------------
+
+Multimodal datasets are currently expected to follow the :ref:`sharegpt` chat format, where the image paths are in one column
+and the user-assistant conversations are in another column.
+
+.. code-block:: text
+
+    |  conversations                     | image        |
+    |------------------------------------|--------------|
+    | [{"from": "human", "value": "Q1"}, | images/1.jpg |
+    |  {"from": "gpt", "value": "A1"}]   |              |
+
+As an example, you can see the schema of the `ShareGPT4V dataset <https://huggingface.co/datasets/Lin-Chen/ShareGPT4V>`_.
```

Comment:
it seems that it forces single image per row, is that right? If so, can we make it explicit that this is a constraint of this format? Otherwise, can we add an example with more than 1 image?

Reply:
yes, will state that it's constrained to single image per row for now